### PR TITLE
allow constructor to take a single hashref

### PR DESCRIPTION
**Pull request by preaction**
The documentation says this works, but it didn't.

### CODE REVIEW

File: lib/Mojo/mysql.pm [hunk 1]
```diff
@@ -7,7 +7,7 @@ use File::Spec::Functions 'file_name_is_absolute';
 use Mojo::mysql::Database;
 use Mojo::mysql::Migrations;
 use Mojo::URL;
-use Scalar::Util 'weaken';
+use Scalar::Util 'weaken', 'blessed';
```
Comment (jhthorsen) [inline lib/Mojo/mysql.pm]: What does `blessed` do? Looks like it's extra..?

File: lib/Mojo/mysql.pm [hunk 2]
```diff
@@ -73,7 +73,11 @@ sub from_string {
   return $self->dsn($dsn);
 }
 
-sub new { @_ == 2 ? shift->SUPER::new->from_string(@_) : shift->SUPER::new(@_) }
+sub new {
+    @_ > 2 || ref $_[-1] eq 'HASH'
+        ? shift->SUPER::new(@_)
+        : shift->SUPER::new->from_string(@_)
```
Comment (jhthorsen) [inline lib/Mojo/mysql.pm]: Please tidy both files, if you want `git blame` to reference you later on 😄

**Review by jhthorsen — CHANGES_REQUESTED** [review]
Wow! What a dumb mistake of me. Thanks for the patch 👍

File: lib/Mojo/mysql.pm (hunk 1 above)
Comment (preaction) [reply lib/Mojo/mysql.pm]: Sorry, yes, as I was iterating to find the solution I could accept, I had imported `blessed` to try to distinguish between Mojo::URL objects and plain hashrefs (which it turns out `ref` handles just fine). I'll remove this.

File: lib/Mojo/mysql.pm (hunk 2 above)
Comment (preaction) [reply lib/Mojo/mysql.pm]: Done!

**Comment by jhthorsen** [comment]
Cool! I’ll try to make a release when I get to work in a couple of hours.

**Pull request merged**